### ISSUE TITLE
ContentPartFieldDefinition Settings.ToObject deserializes to empty object

### ISSUE THREAD

**Issue by davidpuplava** [OrchardCMS/OrchardCore#16764]
### Describe the bug
Content Part Fields Settings are not showing on the edit screen because they are deserialized successfully.

### Orchard Core version
Commit: f9fb26a75daa02b0eee8e5b7688ad33c52af2efc

### To Reproduce
1. Setup default Blog recipe site
2. Go to Content->Content Definition->Content Types
3. Click on Blog Post
4. Scroll down to Fields section
5. Click Edit button next to Subtitle
6. Enter a value for "Hint" such as "Test Hint"
7. Click Save
8. Scroll back down to Fields section
9. Click Edit button next to Subtitle
10. Observe that the Hint text field is empty

### Expected behavior
Expected to see the "Test Hint" value in the "Hint" text field

### Logs and screenshots
Here is a screen capture of the issue (this GIF also shows a potential solution).
![oc-textfield-settings-bug](https://github.com/user-attachments/assets/8837be8b-5999-4981-a689-7922ced11dac)

### Analysis
Looks to be related to the removal of Newtonsoft. The `TextFieldSettingsDriver.cs` file changed to the following block of code to deserialize the text field settings:
```
        return Initialize<TextFieldSettings>("TextFieldSettings_Edit", model =>
        {
            var settings = partFieldDefinition.Settings.ToObject<TextFieldSettings>();

            model.Hint = settings.Hint;
            model.Required = settings.Required;
            model.DefaultValue = settings.DefaultValue;
        }).Location("Content");
```

This particular line of code returns an empty TextFieldSettings object because the Settings property is an array of 2 JsonObjects rather than a TextFieldSettings object directly.
![image](https://github.com/user-attachments/assets/7aae250a-4b5b-4b02-a5fe-9e2a51a91af9)

### Possible Solutions
Inspecting the `Settings` property on the `ContentDefinition` type has a comment that says not to access directly and instead use the GetSettings method. 

Changing to use `GetSettings` method does appear to correctly deserialize the settings object.

### Other Considerations
This looks to affect other field settings such as BooleanFieldSettings. Searching the code base, I see 20 instances of `...Settings.ToObject<...` which likely need to be tested and fixed as well.

Most importantly there are Migrations in `OrchardCore.ContentManagement\Records` that look to access the Settings in the same way. I'm not sure of the implications of changing that migration to use `GetSettings`.

**Comment by davidpuplava** [comment]
I can submit a PR to fix this for deserializing settings in the DisplayDriver classes. But I need guidance on if I should also change to use `GetSettings` in the Migration classes.

**Comment by MikeAlhayek** [comment]
@davidpuplava I think using `.GetSettings<>()` should be the way to go. Can you please confirm that  `.GetSettings<>()` fixes the issue?

If that indeed fixes the issue, you may also want to fix it here

https://github.com/search?q=repo%3AOrchardCMS%2FOrchardCore%20%22.Settings.ToObject%3C%22&type=code

**Comment by davidpuplava** [comment]
Yes, at least in the TextFieldSettings display driver it fixes it. I'll confirm in the other places that there is a problem before I fix it. Thank you @MikeAlhayek.

**Comment by gvkries** [comment]
Oh no, how did we miss this. It's breaking 2.0 and should be fixed asap.

**Comment by davidpuplava** [comment]
I should have the PR ready here shortly, in the next couple of hours. I wrote some unit tests to check each field type.

**Comment by hishamco** [comment]
Waiting for your PR

**Comment by davidpuplava** [comment]
PR submitted: https://github.com/OrchardCMS/OrchardCore/pull/16774
